arch/arm/src/arm/arm_doirq.c

```diff
@@ -71,6 +73,7 @@ uint32_t *arm_doirq(int irq, uint32_t *regs)
    */
 
   up_set_current_regs(regs);
+  tcb->xcp.regs = regs;
```

Comment:
why not mapping tcb->xcp.regs in up_current_regs?

Reply:
tcb->xcp.regs will not be NULL,  how do we determine if we are in irq

In most ARM architectures, we will eventually deprecate the use of g_current_regs.
context switching will directly manipulate tcb->xcp.regs.
To determine if we are in irq, we use per-CPU registers or the interrupt status register.